### PR TITLE
OPS-1761: Adds a Chef pipeline

### PR DESCRIPTION
**Pull request by cmparkinson**

### CODE REVIEW

File: vars/chefPipeline.groovy [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                }
+              }
+            }
+            steps {
+              container('ruby') {
+                sh """
+                  eval `ssh-agent -s`
+                  echo "$GITHUB_SSH_KEY" | ssh-add -
+                  mkdir -p /root/.ssh
+                  ssh-keyscan -t rsa github.com > /root/.ssh/known_hosts
```
Comment (jadametz) [inline vars/chefPipeline.groovy:49]: Is it possible / worth pulling the `ssh-agent` stuff out into a function that just executes that bit of an `sh` step?

File: vars/chefPipeline.groovy [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    stages {
+      stage('Run tests') {
+        parallel {
+          stage('Run unit tests') {
+            agent {
+              kubernetes {
+                label "leroy-unit-${uuid}"
+                containerTemplate {
+                  name 'ruby'
```
Comment (jadametz) [inline vars/chefPipeline.groovy]: I know the Chef CI image is based on Ruby, but if `'ruby'` still an accurate name for this? Just thinking `'chef'` might be more representative.

File: vars/chefPipeline.groovy [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+              }
+            }
+          }
+          stage('Run integration tests') {
+            agent {
+              kubernetes {
+                label "leroy-integration-${uuid}"
+                containerTemplate {
+                  name 'ruby'
+                  image 'invocaops/chef-ci:master'
```
Comment (jadametz) [inline vars/chefPipeline.groovy]: What do you think about pulling the container name and image out into constants at the top of the file? Again, just aiming to DRY things up and treat this like we would most other code.

File: vars/chefPipeline.groovy (hunk 1 above)
Comment (cmparkinson) [reply vars/chefPipeline.groovy:49]: Each `sh` method starts a new shell, which means the environment variables set by eval'ing `ssh-agent -s` don't persist.

File: vars/chefPipeline.groovy (hunk 3 above)
Comment (cmparkinson) [reply vars/chefPipeline.groovy]: Good idea, done.

**Review by jadametz — APPROVED** [review]

**Pull request merged**